File: tests/fixtures/code_intelligence/python_sample/src/sample/producer.py
"""Producer corpus: one protocol, one live implementation, one dead sibling.

The dead sibling is deliberately near-identical to the live implementation so
that a name-only text search cannot distinguish them. Only a semantic backend
that understands imports and types can tell which one the consumer actually
uses.
"""

from __future__ import annotations

from typing import Protocol


class Producer(Protocol):
    """The interface the consumer depends on."""

    def produce(self) -> str: ...


class LiveProducer:
    """The implementation that is actually reachable from the consumer."""

    def produce(self) -> str:
        return "live"


class DeadProducer:
    """Deliberately similar dead sibling. Never imported by the consumer."""

    def produce(self) -> str:
        return "dead"


def make_producer() -> Producer:
    """Live wrapper — the only construction path the consumer imports."""
    return LiveProducer()


def make_dead_producer() -> Producer:
    """Dead wrapper — unreferenced outside this module."""
    return DeadProducer()

File: tests/fixtures/code_intelligence/typescript_sample/src/producer.ts
/**
 * Producer corpus: one interface, one live implementation, one dead sibling.
 *
 * The dead sibling is deliberately near-identical to the live one so that a
 * name-only text search cannot tell them apart. Only a semantic backend that
 * understands imports and types can say which one the consumer reaches.
 */

export interface Producer {
  produce(): string;
}

export class LiveProducer implements Producer {
  produce(): string {
    return "live";
  }
}

export class DeadProducer implements Producer {
  produce(): string {
    return "dead";
  }
}

export function makeProducer(): Producer {
  return new LiveProducer();
}

export function makeDeadProducer(): Producer {
  return new DeadProducer();
}
